### DISCUSSION TITLE
ivabs license failure message

### DISCUSSION THREAD

**Discussion by joonwlim** [wenbinyugroup/.github#20] [VABS Related Discussions] [answered]
Ran a sample case by issuing 'python run.py  cs_param_study.yml'. I am getting the error message as below. How can I fix it?

****** 
[INFO] running analysis...
[CRITICAL] VABS cs_1.sg
This is VABS 4.1 compiled at 15:29:13 of Aug 01 2024.

 
 Homogenization for computing sectional properties

 The inputs are echoed in cs_1.sg.ech

 Finished reading inputs for the cross-sectional analysis.
  Something wrong with the license:  Can't connect to license server, xxxxx

**Comment by wenbinyu-PU** [comment]
If you use a floating license, please make sure the license server is running with a valid license. Your computer can talk to the server. and your computer is configurated with the environmental variable: SG_LICENSE_FILE. Please refer to quick start file.

**Reply by joonwlim** [reply]
> If you use a floating license, please make sure the license server is running with a valid license. Your computer can talk to the server. and your computer is configurated with the environmental variable: SG_LICENSE_FILE. Please refer to quick start file.

No, I am currently using a node-lock license. How can I change it to the node-locked license?

**Reply by wenbinyu-PU** [reply]
I just emailed you. Basically, you need to issue the command hostname to report the real hostname and make sure it matches the hostname in the license file. And make sure the license file is in the same folder as the executable VABS.

**Comment by joonwlim** [comment]
$ hostname -f
t-aax-limjw.ds.amrdec.army.mil

Hostname was provided as above, but there were hostname name changes a few times. Last time, hostname was t-aax-limj instead of limjw. So, I am not sure whether all the confusions on hostname  in my linux box were cleared. 

I tried Ravis' floating license, and it worked. In that license file, the domain name (ds.amrdec.army.mil) was nto provided but it worked. I wonder whether it may be related to the domain name. Thanks.

**Reply by wenbinyu-PU** [reply]
Great. It could the hostname is too long. I am glad that the floating license worked for you. Then you can just use the floating license.

**Comment by joonwlim** [comment]
Wenbin,

With Ravi's floating license, ivabs v0.8 seems running now in my linux box. I like to upgrade to v0.11. In the repository, there are six assets. We are running Redhat instead of Centos. Which assets do you suggest to download - 1) between centos installer vs. portable, 2) Examples, 3) what the source codes serve for?

Thanks,

Joon

**Comment by joonwlim** [comment]
With Ravi’s floating license, ivabs v0.8 seems running now in my linux box. I like to upgrade to v0.11. In the repository, there are six assets. We are running Redhat instead of Centos. Which assets do you suggest to download – 1) between centos installer vs. portable, 2) Examples, 3) what the source codes serve for?  Thanks.

**Reply by wenbinyu-PU** [reply]
I will let @unitasium answer this question.

**Reply by unitasium** [reply]
You can try the centos portable version first. If that does not work, we can do two options: 1) I will find a way to build the code for rhel (like a subsystem on windows, or let github to do this), or 2) I will instruct you to build the code on your machine. I think the only code that matters is prevabs. The rest are mainly python.

**Reply by joonwlim** [reply]
So, how different between centos installer and centos portable? When do people need the asset, source?

**Reply by unitasium** [reply, marked as answer]
The installer provides a simple command line interface to guide you through the installation process including unpacking and setting the system variables, while the portable one requires you do everything manually. Actually, you can try the installer first. The source assets contain nothing. They are created automatically when creating a new release. The ivabs repository is used only for storing the released package and hosting the online documentation.
